### PR TITLE
[#125] Split template into multiple modules

### LINKED ISSUE #125: Split `Summoner.Template` into smaller pieces
This is a first step towards refactoring to `update` command. We're going to use `named` library for this:

* https://hackage.haskell.org/package/named-0.2.0.0/docs/Named.html

### PR DESCRIPTION
**Pull request by ChShersh**
Resolves #125

Huge PR with massive refactoring. TBD: test manually under different conditions.

## Checklist:

- [x] I've updated the [CHANGELOG](https://github.com/kowainik/summoner/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [x] All new and existing tests pass.
- [x] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [x] I've used the [`stylish-haskell` file](https://github.com/kowainik/summoner/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.
- [x] I changed code related to the generated project.
  - [ ] I created a new project using `summoner` and I checked that this specific feature is working as expected.

### CODE REVIEW

**Comment by ChShersh** [comment]
@vrom911 I've created 3 different projects, looked at them manually (the content looks valid) and put the last one on GitHub. The CI seems to pass

* https://github.com/ChShersh/test-github

File: src/Summoner/Template.hs [hunk 1]
```diff
@@ -1,644 +1,34 @@
-{-# LANGUAGE QuasiQuotes  #-}
-{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE OverloadedLabels #-}
```
Comment (vrom911) [inline src/Summoner/Template.hs]: I guess this is no needed anymore

File: src/Summoner/Template.hs [hunk 2]
```diff
@@ -1,644 +1,34 @@
...
-        # The ugly echo "" hack is to avoid complaints about 0 being an invalid file
-        # descriptor
-        - echo "" | stack --arch x86_64 --no-terminal build --bench --no-run-benchmarks --test
-        |]
+    cabal, stack :: [TreeFs]
+    cabal   = [cabalFile settings]
+    stack   = memptyIfFalse settingsStack $ stackFiles settings  -- TODO: write more elegant
+    haskell = haskellFiles settings
+    docs    = docFiles settings
+    gitHub = gitHubFiles settings
```
Comment (vrom911) [inline src/Summoner/Template.hs]:
```suggestion
    gitHub  = gitHubFiles settings
```

File: src/Summoner/Template/Cabal.hs [hunk 3]
```diff
@@ -0,0 +1,203 @@
+{-# LANGUAGE DataKinds        #-}
+{-# LANGUAGE OverloadedLabels #-}
```
Comment (vrom911) [inline src/Summoner/Template/Cabal.hs]: And here as well 🙂

**Pull request merged**